### PR TITLE
Soft-deleting times

### PR DESCRIPTION
**Pull request by ElijahCaine**
These are the initial tests for soft-deleting times (DELETE ``/times/:uuid``)

### CODE REVIEW

File: tests/times.js [hunk 1]
```diff
@@ -3283,4 +3283,56 @@ module.exports = function(expect, request, baseUrl) {
...
+              created_at: '2015-04-19',
+              updated_at: null,
+              deleted_at: null,
+              uuid: '32764929-1bea-4a17-8c8a-22d7fb144941',
+              revision: 1,
+              id: 1,
+            },
+          ];
+
+          // TODO: GET should only return 200 when ?revisions=true is passed.
```
Comment (Kennric) [inline tests/times.js]: Shouldn't this test return 404, since we deleted the object?

**Comment by Kennric** [comment]
That looks better. +1

**Comment by LyonesGamer** [comment]
I would also add a test for sending an invalid UUID (i.e. a string that can't even be parsed as a UUID), which would return an Invalid Identifier error instead of an Object Not Found.

**Comment by LyonesGamer** [comment]
+1 Looks good to me.

**Comment by Kennric** [comment]
I think that's got it. +1

File: src/times.js [hunk 2]
```diff
@@ -793,4 +793,45 @@ module.exports = function(app) {
...
+    knex('times').select('id').where('uuid', req.params.uuid).first()
+    .then(function(time) {
+      if (!time) {
+        const err = errors.errorObjectNotFound('uuid', req.params.uuid);
+        return res.status(err.status).send(err);
+      }
+
+      knex.transaction(function(trx) {
+        trx('times').where('uuid', req.params.uuid)
+        .update({'deleted_at': Date.now(), 'uuid': null})
```
Comment (LyonesGamer) [inline src/times.js]: Why are we setting ``uuid`` to null?

File: src/times.js [hunk 3]
```diff
@@ -793,4 +793,45 @@ module.exports = function(app) {
...
+
+    knex('times').select('id').where('uuid', req.params.uuid).first()
+    .then(function(time) {
+      if (!time) {
+        const err = errors.errorObjectNotFound('uuid', req.params.uuid);
+        return res.status(err.status).send(err);
+      }
+
+      knex.transaction(function(trx) {
+        trx('times').where('uuid', req.params.uuid)
```
Comment (LyonesGamer) [inline src/times.js]: I think we only want to be setting ``deleted_at`` on the "current" revision.

File: src/times.js [hunk 4]
```diff
@@ -793,4 +793,45 @@ module.exports = function(app) {
...
+
+    knex('times').select('id').where('uuid', req.params.uuid).first()
+    .then(function(time) {
+      if (!time) {
+        const err = errors.errorObjectNotFound('uuid', req.params.uuid);
+        return res.status(err.status).send(err);
+      }
+
+      knex.transaction(function(trx) {
+        trx('times').where('uuid', req.params.uuid).first()
```
Comment (LyonesGamer) [inline src/times.js]: Because order isn't guaranteed on a database call, you should impose order here, ``.orderBy('revision', 'desc')``

**Comment by LyonesGamer** [comment]
+1

**Comment by Kennric** [comment]
Looks good. +1

**Pull request merged**